src/collector/top_field_collector.rs

```diff
@@ -0,0 +1,267 @@
+use collector::TopCollector;
+use DocAddress;
+use DocId;
+use fastfield::FastFieldReader;
+use fastfield::FastValue;
+use Result;
+use Score;
+use SegmentReader;
+use super::Collector;
+use TantivyError;
+
+/// The Top Field Collector keeps track of the K documents
+/// sorted by a fast field in the index
+///
+/// The implementation is based on a `BinaryHeap`.
+/// The theorical complexity for collecting the top `K` out of `n` documents
+/// is `O(n log K)`.
+///
+/// ```rust
+/// #[macro_use]
+/// extern crate tantivy;
+/// use tantivy::schema::{SchemaBuilder, TEXT, FAST};
+/// use tantivy::{Index, Result, DocId};
+/// use tantivy::collector::TopFieldCollector;
+/// use tantivy::query::QueryParser;
+///
+/// # fn main() { example().unwrap(); }
+/// fn example() -> Result<()> {
+///     let mut schema_builder = SchemaBuilder::new();
+///     let title = schema_builder.add_text_field("title", TEXT);
+///     let rating = schema_builder.add_u64_field("rating", FAST);
+///     let schema = schema_builder.build();
+///     let index = Index::create_in_ram(schema);
+///     {
+///         let mut index_writer = index.writer_with_num_threads(1, 3_000_000)?;
+///         index_writer.add_document(doc!(
+///             title => "The Name of the Wind",
+///             rating => 92u64,
+///         ));
+///         index_writer.add_document(doc!(
+///             title => "The Diary of Muadib",
+///             rating => 97u64,
+///         ));
+///         index_writer.add_document(doc!(
+///             title => "A Dairy Cow",
+///             rating => 63u64,
+///         ));
+///         index_writer.add_document(doc!(
+///             title => "The Diary of a Young Girl",
+///             rating => 80u64,
+///         ));
+///         index_writer.commit().unwrap();
+///     }
+///
+///     index.load_searchers()?;
+///     let searcher = index.searcher();
+///
+///     {
+///	        let mut top_collector = TopFieldCollector::with_limit("rating", 2);
```

Comment:
indent

Reply:
Strange, it isn't like this in my workspace.